vllm/model_executor/models/llama.py

```diff
@@ -386,6 +386,52 @@ def sample(
         next_tokens = self.sampler(logits, sampling_metadata)
         return next_tokens
 
+    def _convert_weights_to_fp8(
+            self, state_dict: Dict[str, torch.tensor],
+            weights_to_convert: List[str]) -> Dict[str, torch.tensor]:
+        """Converts the original weights to FP8E4M3 from FP16."""
+        for weight_name in weights_to_convert:
+            weight_scale_name = weight_name + "_scale"
+            if weight_scale_name not in state_dict:
+                continue
+            loaded_weight = state_dict[weight_name]
+            scale = state_dict[weight_scale_name]
+            state_dict[weight_name] = (loaded_weight.cpu() / scale.cpu()).to(
+                torch.float8_e4m3fn)
+        return state_dict
+
+    def _convert_scales_for_vllm(self, key, value):
+        """Replaces the names of *quantizer._amax to _scale."""
+        replacements = {
+            "weight_quantizer._amax": "weight_scale",
+            "input_quantizer._amax": "act_scale",
+        }
+        for old_suffix, new_suffix in replacements.items():
+            if key.endswith(old_suffix):
+                new_key = key[:len(key) - len(old_suffix)] + new_suffix
+                new_value = value / 448
+                return new_key, new_value
+        else:
+            return key, value
+
+    def _convert_ammo_weights(self, input_state_dict: Dict[str, torch.tensor]):
+        """Util method to modify the modelopt state dict to vLLM checkpoint."""
+        weights_to_convert = []
+        vllm_state_dict = {}
+        for key, value in input_state_dict.items():
+            if key.endswith("_amax"):
```

Comment:
It would be best to make this as specific as possible to avoid possible conflicts -- would `if key.endswith("_quantizer._amax"):` work?